libraries/Filter/NotchFilter.cpp

```diff
@@ -71,7 +71,7 @@ void NotchFilter<T>::init_with_A_and_Q(float sample_freq_hz, float center_freq_h
                                           _center_freq_hz * NOTCH_MAX_SLEW_UPPER);
     }
 
-    if ((new_center_freq > 0.0) && (new_center_freq < 0.5 * sample_freq_hz) && (Q > 0.0)) {
+    if ((new_center_freq > 2.0) && (new_center_freq < 0.5 * sample_freq_hz) && (Q > 0.0)) {
```

Comment:
that change was in a previous PR that has already been approved and merged (though @andyp1per objected to it after the merge)
The reason I like the 2Hz check is this effort came out of a flight where a quadplane ended up with a 1Hz notch filter, which caused it to crash. The reason we ended up with the 1Hz filter was the fact that the initial freq was only done for the first ESC source, but the code is complex enough I am not absolutely confident there is not another code path that can lead to a low filter like that. It ended up as 1Hz not 0Hz due to the constrain_float() in HarmonicNotchFilter. 
For the 4.4.x backport I've chosen to go with the 2.0Hz check as it is the simplest to reason about for a stable bugfix release. We never want notches down that low, so changing the 0.0Hz to 2.0Hz is a guaranteed way that there is no code path that will result in these low filter frequencies.
For 4.5.x the situation is different and I would prefer to see us fix the way that the frequency limits are currently spread all over the code.
The reason I don't think Pete's change is correct is it is filling in the wrong variable.
The array notch.calculated_notch_freq_hz[] is a set of measurements from the ESCs. By doing the setting of that measurement to the min frequency on initialisation we are providing false data to the HarmonicNotch filter. We're telling it "we have seen this frequency from an ESC" when we have not seen that at all. That means if there is never any update from an ESC (which there never was in the crash report that triggered this work) then we are leaving the filter incorrectly initialised with a false value, and that prevents us properly handling the zero frequency case.
What we should be doing is *not* initialising, leaving the array as zero (meaning uninitialised) then handling the zero in the HarmonicNotch code, with whatever mechanism we deliberately decide on.
Alternatively we could fill in with NaN meaning uninitialised, or -1, but I think zero is good and works well.

Reply:
This is nothing more than a bug and it should never have been merged. The PR you are referring too didn't fix a bug, it added a bug that disguised the original bug. That is two steps backwards.

> It ended up as 1Hz not 0Hz due to the constrain_float() in HarmonicNotchFilter.

Yes, this is the bug that needed to be fixed.

> For the 4.4.x backport I've chosen to go with the 2.0Hz check as it is the simplest to reason about for a stable bugfix release. We never want notches down that low, so changing the 0.0Hz to 2.0Hz is a guaranteed way that there is no code path that will result in these low filter frequencies.

Yes we do want notches to be used down low! This is one of the key reasons we used a peaking filter rather than a notch filter. By preventing the use of the filter as it was designed to be used you have fundamentally changed the behaviour of the code. And on top of that a 2 Hz filter with lots of attenuation is just as likely to cause problems as a 1 Hz notch filter. There is no justification you can give for choosing 2 Hz over 1 Hz, or 3 Hz, maybe 10 Hz.

> For the 4.4.x backport

There is no way we can put that PR back in 4.4.x.